drivers/input/input_sbus.c

```diff
@@ -0,0 +1,363 @@
+/*
+ * Copyright (c) 2024 CogniPilot Foundation
+ * Copyright (c) 2024 NXP Semiconductors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT futaba_sbus
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/input/input.h>
+#include <zephyr/irq.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/time_units.h>
+#include <zephyr/sys_clock.h>
+#include <zephyr/drivers/uart.h>
+
+LOG_MODULE_REGISTER(futaba_sbus, CONFIG_INPUT_LOG_LEVEL);
+
+/* Driver config */
+struct sbus_input_channel {
+	uint32_t sbus_channel;
+	uint32_t type;
+	uint32_t zephyr_code;
+};
+
+const struct uart_config uart_cfg_sbus = {
+	.baudrate = 100000,
+	.parity = UART_CFG_PARITY_EVEN,
+	.stop_bits = UART_CFG_STOP_BITS_2,
+	.data_bits = UART_CFG_DATA_BITS_8,
+	.flow_ctrl = UART_CFG_FLOW_CTRL_NONE
+};
+
+struct input_sbus_config {
+	uint8_t num_channels;
+	const struct sbus_input_channel *channel_info;
+	const struct device *uart_dev;
+	uart_irq_callback_user_data_t cb;
+};
+
+#define SBUS_FRAME_LEN 25
+#define SBUS_HEADER    0x0F
+#define SBUS_FOOTER    0x00
+
+#define SBUS_SERVO_LEN     22
+#define SBUS_SERVO_CH_MASK 0x7FF
+
+#define SBUS_BYTE24_IDX        23
+#define SBUS_BYTE24_CH17       0x01
+#define SBUS_BYTE24_CH18       0x02
+#define SBUS_BYTE24_FRAME_LOST 0x04
+#define SBUS_BYTE24_FAILSAFE   0x08
+
+#define SBUS_TRANSMISSION_TIME  4  /* 4ms, Max transmission of a single SBUS frame */
+#define SBUS_INTERFRAME_SPACING 20 /* 20ms, Max spacing between SBUS frames */
```

Comment:
```suggestion
#define SBUS_TRANSMISSION_TIME_MS  4  /* Max transmission of a single SBUS frame */
#define SBUS_INTERFRAME_SPACING_MS 20 /* Max spacing between SBUS frames */
```

(actually I would drop the comments entirely, up to you)